datafusion/expr/src/udf.rs

```diff
@@ -174,6 +174,7 @@ impl ScalarUDF {
     ///
     /// See [`ScalarUDFImpl::return_type`] for more details.
     pub fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
+        #[allow(deprecated)]
```

Comment:
Shouldn't we also deprecate this (`ScalarUDF::return_type`) method if we deprecate underlying `ScalarUDFImpl` method?

Reply:
Yes